--- libcxx/include/stdatomic.h.old	2023-12-06 21:01:46.168049453 +0800
+++ libcxx/include/stdatomic.h	2023-12-06 21:01:55.056057032 +0800
@@ -220,7 +220,7 @@
 using std::atomic_signal_fence _LIBCPP_USING_IF_EXISTS;
 using std::atomic_thread_fence _LIBCPP_USING_IF_EXISTS;
 
-#elif defined(_LIBCPP_COMPILER_CLANG_BASED)
+#else
 
 // Before C++23, we include the next <stdatomic.h> on the path to avoid hijacking
 // the header. We do this because Clang has historically shipped a <stdatomic.h>
