--- a/include/__ostream/basic_ostream.h
+++ b/include/__ostream/basic_ostream.h
@@ -207,7 +207,12 @@

 template <class _CharT, class _Traits>
 basic_ostream<_CharT, _Traits>::sentry::~sentry() {
-  if (__os_.rdbuf() && __os_.good() && (__os_.flags() & ios_base::unitbuf) && uncaught_exceptions() == 0) {
+  if (__os_.rdbuf() && __os_.good() && (__os_.flags() & ios_base::unitbuf)
+#  if _LIBCPP_HAS_EXCEPTIONS
+      && uncaught_exceptions() == 0
+#  endif
+      ) {
+
 #  if _LIBCPP_HAS_EXCEPTIONS
     try {
 #  endif // _LIBCPP_HAS_EXCEPTIONS
