--- libcxx/src/locale.cpp.orig
+++ libcxx/src/locale.cpp
@@ -5605,6 +5605,7 @@
 
 // explicit instantiation

+#ifndef _LIBCPP_HAS_NO_LOCALIZATION
 template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS collate<char>;
 template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS collate_byname<char>;
 template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS moneypunct_byname<char, false>;
@@ -5645,6 +5646,7 @@
 template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS collate_byname<wchar_t>;
 template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS codecvt_byname<wchar_t, char, mbstate_t>;
 #endif
+#endif // _LIBCPP_HAS_NO_LOCALIZATION
 
 // FIXME: These explicit instantiations seem wrong since it depends on localization support which can be disabled
 _LIBCPP_SUPPRESS_DEPRECATED_PUSH
