--- a/src/string.cpp	2026-01-10 09:53:40.035509375 +0100
+++ b/src/string.cpp	2026-01-10 09:53:40.045300744 +0100
@@ -382,4 +382,13 @@
 wstring to_wstring(long double val) { return as_string(get_swprintf(), initial_string<wstring>()(), L"%Lf", val); }
 #endif
 
+// Explicit template instantiations for embedded systems
+template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS basic_string<char>;
+#if _LIBCPP_HAS_WIDE_CHARACTERS
+template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS basic_string<wchar_t>;
+#endif
+#ifndef _LIBCPP_HAS_NO_CHAR8_T
+template class _LIBCPP_CLASS_TEMPLATE_INSTANTIATION_VIS basic_string<char8_t>;
+#endif
+
 _LIBCPP_END_NAMESPACE_STD
