--- libs/libxx/libcxx/include/__mbstate_t.h	2023-11-28 09:52:28.000000000 +0100
+++ libs/libxx/libcxx/include/__mbstate_t.h	2023-12-05 16:14:46.523689408 +0100
@@ -39,12 +39,12 @@
 #   define __NEED_mbstate_t
 #   include <bits/alltypes.h>
 #   undef __NEED_mbstate_t
+#elif !defined(_LIBCPP_HAS_NO_WIDE_CHARACTERS) && __has_include_next(<wchar.h>)
+#   include_next <wchar.h> // fall back to the C standard provider of mbstate_t
 #elif __has_include(<bits/types/mbstate_t.h>)
 #   include <bits/types/mbstate_t.h> // works on most Unixes
 #elif __has_include(<sys/_types/_mbstate_t.h>)
 #   include <sys/_types/_mbstate_t.h> // works on Darwin
-#elif !defined(_LIBCPP_HAS_NO_WIDE_CHARACTERS) && __has_include_next(<wchar.h>)
-#   include_next <wchar.h> // fall back to the C standard provider of mbstate_t
 #elif __has_include_next(<uchar.h>)
 #   include_next <uchar.h> // <uchar.h> is also required to make mbstate_t visible
 #else
