--- libcxxabi.orig/src/cxa_guard_impl.h	2026-01-10 14:08:51.808835600 +0100
+++ libcxxabi/src/cxa_guard_impl.h	2026-01-10 14:08:51.819473598 +0100
@@ -665,7 +665,7 @@
 // TODO(EricWF): We should prefer the futex implementation when available. But
 // it should be done in a separate step from adding the implementation.
 constexpr Implementation CurrentImplementation =
-#if defined(_LIBCXXABI_HAS_NO_THREADS)
+#if defined(_LIBCXXABI_HAS_NO_THREADS) || defined(_LIBCPP_HAS_NO_THREADS)
     Implementation::NoThreads;
 #elif defined(_LIBCXXABI_USE_FUTEX)
     Implementation::Futex;

